aiida_common_workflows/cli/launch.py

```diff
@@ -78,3 +79,131 @@ def cmd_relax(plugin, structure, protocol, relaxation_type, threshold_forces, th
+            schema = generator.get_calc_type_schema(engine)
+            click.secho(engine, fg='red', bold=True)
+            click.echo('Required code plugin: {}'.format(schema['code_plugin']))
+            click.echo('Engine description:   {}'.format(schema['description']))
+
+        return
+
+    engines = {}
+
+    for engine in generator.get_calc_types():
+        schema = generator.get_calc_type_schema(engine)
+        engines[engine] = {
+            'options': {
+                'resources': {
+                    'num_machines': 1
+                },
+                'max_wallclock_seconds': 86400,
+            }
+        }
+        code_plugin = schema['code_plugin']
+        builder = QueryBuilder().append(Code, filters={'attributes.input_plugin': code_plugin})
+
+        code = builder.first()
+
+        if code is None:
+            raise click.UsageError(f'could not find a configured code for the plugin `{code_plugin}`.')
+
+        engines[engine]['code'] = code[0]
+
+    builder = generator.get_builder(structure, engines, protocol, relaxation_type, threshold_forces, threshold_stress)
+    inputs = {
+        'structure': structure,
+        'sub_process_class': get_entry_point_name_from_class(process_class).name,
+        'sub_process': {
+            **builder._inputs(prune=True)  # pylint: disable=protected-access
+        }
+    }
+    inputs['sub_process'].pop('structure', None)
+    utils.launch_process(EquationOfStateWorkChain, daemon, **inputs)
+
+
+@cmd_launch.command('plot-eos')
+@options_core.NODE()
+def cmd_plot_eos(node):
+    """Plot the results from an `EquationOfStateWorkChain`."""
+    # pylint: disable=too-many-locals,invalid-name
+    import numpy
+    import pylab as plt
+
+    from aiida.common import LinkType
+
+    def birch_murnaghan(V, E0, V0, B0, B01):
+        """Compute energy by Birch Murnaghan formula."""
+        r = (V0 / V)**(2. / 3.)
+        return E0 + 9. / 16. * B0 * V0 * (r - 1.)**2 * (2. + (B01 - 4.) * (r - 1.))
+
+    def fit_birch_murnaghan_params(volumes, energies):
+        """Fit Birch Murnaghan parameters."""
+        from scipy.optimize import curve_fit
+
+        params, covariance = curve_fit(  # pylint: disable=unbalanced-tuple-unpacking
+            birch_murnaghan,
+            xdata=volumes,
+            ydata=energies,
+            p0=(
+                energies.min(),  # E0
+                volumes.mean(),  # V0
+                0.1,  # B0
+                3.,  # B01
+            ),
+            sigma=None
+        )
+        return params, covariance
+
+    outputs = node.get_outgoing(link_type=LinkType.RETURN).nested()
+
+    volumes = []
+    energies = []
+    for index, structure in sorted(outputs['structures'].items()):
+        volumes.append(structure.get_cell_volume())
+        energies.append(outputs['total_energies'][index].value)
+
+    params, _ = fit_birch_murnaghan_params(numpy.array(volumes), numpy.array(energies))
+
+    vmin = volumes.min()
```

Comment:
Weird, I ran this on Quantum Mobile, but looking at this code it should indeed always have failed. I will try to add a unit test to prevent this from happening

Reply:
Same thing here. Volumes is a list. 
```
volumes = numpy.array(volumes)
energies = numpy.array(energies)
params, _ = fit_birch_murnaghan_params(volumes, energies)
```
This solves the issue